### ISSUE TITLE
Steam Art Manager doesn't recognize some games

### ISSUE THREAD

**Issue by mesterjani** [Tormak9970/Steam-Art-Manager#80]
I have 48 games that don't appear in the application. Most of them seem to be games that are unavailable through the Steam Store, but I have a lot that appears despite being delisted.
The notable ones:
- ADR1FT
- Dead Island
- Dead Island: Riptide
- Dirt: Showdown
- Duke Nukem 3D: Megaton Edition
- GRID (Race Driver: Grid)
- Hitman: Sniper Challange
- System Shock Classic

Both of the API Keys seem to be saved properly in the configuration file in the AppData folder.

```
[Steam Art Manager] [22:18:14] [INFO]: Initialized logging file
[Steam Art Manager] [22:18:14] [INFO]: Added Steam directory to scope.
[Steam Art Manager] [22:18:14] [INFO]: Initializing CacheController...
[Steam Art Manager] [22:18:14] [INFO]: Checking config/loginusers.vdf for current user info.
[Steam Art Manager] [22:18:14] [INFO]: Loaded 2 steam users.
[Steam Art Manager] [22:18:14] [INFO]: Found cache dir.
[Steam Art Manager] [22:18:14] [INFO]: Found grids cache dir.
[Steam Art Manager] [22:18:14] [INFO]: App setup complete.
[Steam Art Manager] [22:18:14] [INFO]: App loaded. IsOnline: true.
[Steam Art Manager] [22:18:14] [INFO]: Loading non-steam games...
[Steam Art Manager] [22:18:14] [INFO]: Getting steam shortcuts.vdf...
[Steam Art Manager] [22:18:14] [INFO]: shortcuts.vdf exists, reading...
[Steam Art Manager] [22:18:14] [INFO]: Found Capsule cache dir.
[Steam Art Manager] [22:18:14] [INFO]: Loaded non-steam games.
[Steam Art Manager] [22:18:14] [INFO]: Getting steam games...
[Steam Art Manager] [22:18:14] [INFO]: Getting steam grids folder...
[Steam Art Manager] [22:18:14] [INFO]: Found Wide Capsule cache dir.
[Steam Art Manager] [22:18:14] [INFO]: Found Hero cache dir.
[Steam Art Manager] [22:18:14] [INFO]: Found Logo cache dir.
[Steam Art Manager] [22:18:14] [INFO]: Found Icon cache dir.
[Steam Art Manager] [22:18:14] [INFO]: Initialized CacheController.
[Steam Art Manager] [22:18:14] [INFO]: Grids loaded.
[Steam Art Manager] [22:18:14] [INFO]: Getting steam library cache folder...
[Steam Art Manager] [22:18:15] [INFO]: Library Cache loaded.
[Steam Art Manager] [22:18:15] [INFO]: Loading games from Steam Community page...
[Steam Art Manager] [22:18:18] [INFO]: Loaded 880 games from Steam Community page.
[Steam Art Manager] [22:18:18] [INFO]: Steam games loaded.
```

**System information:**
 - OS: Windows 10
 - Version 2.1.2
 - Steam Beta
![kép](https://github.com/Tormak9970/Steam-Art-Manager/assets/19255274/47f13ab1-37b7-4fa5-9a8e-a1148d51e6b4)

**Comment by Tormak9970** [comment]
> I have 48 games that don't appear in the application. Most of them seem to be games that are unavailable through the Steam Store, but I have a lot that appears despite being delisted. The notable ones:
> 
>     * ADR1FT
> 
>     * Dead Island
> 
>     * Dead Island: Riptide
> 
>     * Dirt: Showdown
> 
>     * Duke Nukem 3D: Megaton Edition
> 
>     * GRID (Race Driver: Grid)
> 
>     * Hitman: Sniper Challange
> 
>     * System Shock Classic
> 
> 
> Both of the API Keys seem to be saved properly in the configuration file in the AppData folder.
> 
> ```
> [Steam Art Manager] [22:18:14] [INFO]: Initialized logging file
> [Steam Art Manager] [22:18:14] [INFO]: Added Steam directory to scope.
> [Steam Art Manager] [22:18:14] [INFO]: Initializing CacheController...
> [Steam Art Manager] [22:18:14] [INFO]: Checking config/loginusers.vdf for current user info.
> [Steam Art Manager] [22:18:14] [INFO]: Loaded 2 steam users.
> [Steam Art Manager] [22:18:14] [INFO]: Found cache dir.
> [Steam Art Manager] [22:18:14] [INFO]: Found grids cache dir.
> [Steam Art Manager] [22:18:14] [INFO]: App setup complete.
> [Steam Art Manager] [22:18:14] [INFO]: App loaded. IsOnline: true.
> [Steam Art Manager] [22:18:14] [INFO]: Loading non-steam games...
> [Steam Art Manager] [22:18:14] [INFO]: Getting steam shortcuts.vdf...
> [Steam Art Manager] [22:18:14] [INFO]: shortcuts.vdf exists, reading...
> [Steam Art Manager] [22:18:14] [INFO]: Found Capsule cache dir.
> [Steam Art Manager] [22:18:14] [INFO]: Loaded non-steam games.
> [Steam Art Manager] [22:18:14] [INFO]: Getting steam games...
> [Steam Art Manager] [22:18:14] [INFO]: Getting steam grids folder...
> [Steam Art Manager] [22:18:14] [INFO]: Found Wide Capsule cache dir.
> [Steam Art Manager] [22:18:14] [INFO]: Found Hero cache dir.
> [Steam Art Manager] [22:18:14] [INFO]: Found Logo cache dir.
> [Steam Art Manager] [22:18:14] [INFO]: Found Icon cache dir.
> [Steam Art Manager] [22:18:14] [INFO]: Initialized CacheController.
> [Steam Art Manager] [22:18:14] [INFO]: Grids loaded.
> [Steam Art Manager] [22:18:14] [INFO]: Getting steam library cache folder...
> [Steam Art Manager] [22:18:15] [INFO]: Library Cache loaded.
> [Steam Art Manager] [22:18:15] [INFO]: Loading games from Steam Community page...
> [Steam Art Manager] [22:18:18] [INFO]: Loaded 880 games from Steam Community page.
> [Steam Art Manager] [22:18:18] [INFO]: Steam games loaded.
> ```
> 
> **System information:**
> 
>     * OS: Windows 10
> 
>     * Version 2.1.2
> 
>     * Steam Beta
>       ![kép](https://user-images.githubusercontent.com/19255274/238155763-47f13ab1-37b7-4fa5-9a8e-a1148d51e6b4.png)

Hmm, the log says it used the steam community page to load your games, not surprised some unlisted games arent showing up. Can you try disabling your wifi/internet connection and opening Steam Art Manager? It will prompt about being offline and you can continue. Then it will use steam's files to load your games. Let me know if your games show up there

**Comment by mesterjani** [comment]
I've been looking at a blank window for 4 hours now. I'm not sure if it's ever going to load but I can't wait longer.
I have Steam closed and SAM is definitely doing something CPU related according to the Task Manager.
I'm guessing your advice would work but then I'm running into the problem with this method being extremely slow with large libraries.
```
[Steam Art Manager] [1:34:39] [INFO]: Initialized logging file
[Steam Art Manager] [1:34:39] [INFO]: Added Steam directory to scope.
```

**Comment by Tormak9970** [comment]
Shoot, didn't think about that. Have you tried adding a Steam API key?

**Comment by mesterjani** [comment]
Yes. I have the Steam API key saved correctly in the settings.json file. I tried it with and without but the results are the same as my original issue.
I have noticed however that in the app the Steam API key is not visible even if it's saved correctly in the settings file. Does this mean that it's not using the API properly and therefore can't load all the games? When I paste the Steam API key into the field and press reload the results are the same. When I exit the app and launch it again, the Steam API key field is empty despite it being present in the settings file.
![kép](https://github.com/Tormak9970/Steam-Art-Manager/assets/19255274/7c334b48-e8e4-4593-99ef-98bd815f034e)
